src/background/icon-manager.ts

```diff
@@ -66,9 +74,34 @@ export default class IconManager {
             return;
         }
         IconManager.iconState.active = false;
-        chrome.browserAction.setIcon({
-            path: IconManager.ICON_PATHS.inactive,
-        });
+
+        this.setTheme(UserStorage.settings.theme.mode);
+    }
+
+    static setTheme(theme: FilterMode): void {
+        if (__THUNDERBIRD__ || !chrome.browserAction.setIcon) {
+            // Fix for Firefox Android and Thunderbird.
+            return;
+        }
+
+        if(IconManager.iconState.active) {
+            chrome.browserAction.setIcon({
```

Comment:
Try calling the `chrome.browserAction.setIcon` once outside of `if-else`. Assign `path` variable in `if-else` or `x ? y : z` depending on what will be more readable.